src/scribe.js

```diff
@@ -71,8 +72,10 @@ define([
     var TransactionManager = buildTransactionManager(this);
     this.transactionManager = new TransactionManager();
 
-    var UndoManager = buildUndoManager(this);
-    this.undoManager = new UndoManager();
+    if (this.options.shouldUndo) {
+      var UndoManager = buildUndoManager(this);
+      this.undoManager = new UndoManager();
```

Comment:
Oh I didn't notice that. Could the presence of `scribe.undoManager` be used in conditions instead of `scribe.options.shouldUndo` in that case? That would reduce the API surface the code has to know about (test for the presence of the thing you're about to conditionally use, rather than for a flag that you know implicitly determines whether that thing you're about to conditionally use has been set).